### PR TITLE
Feature: todos

### PR DESCRIPTION
**Pull request by yarre-uk**

### CODE REVIEW

File: src/pages/todos/components/ToDoItem.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const ToDoItem = ({ todo }: IToDoItem) => {
+  return (
+    <ToDoItemStyled>
+      <p>{todo.id}</p>
+      <h4>{todo.todo}</h4>
+    </ToDoItemStyled>
+  );
+};
+
+const ToDoItemStyled = styled.li`
```
Comment (DmitriyGo) [inline src/pages/todos/components/ToDoItem.tsx:18]: Сreate a new folder for each component and separate the content into logic and styles
The style file should be named like this - ToDoItemStyles.ts

Comment (DmitriyGo) [reply src/pages/todos/components/ToDoItem.tsx:18]: Styled component name should start with Styled
StyledToDoItem

File: src/pages/todos/index.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    flex-direction: column;
+    gap: 10px;
+  }
+`;
+
+export const getServerSideProps: GetServerSideProps =
+  wrapper.getServerSideProps((store) => async (context) => {
+    const { locale = 'uk', res } = context;
+
+    const asd = store.dispatch(todoApi.endpoints.getToDos.initiate());
```
Comment (DmitriyGo) [inline src/pages/todos/index.tsx:43]: No more 'asd' even in temporary files

File: src/pages/todos/index.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    const data = (await asd).data;
```
Comment (DmitriyGo) [inline src/pages/todos/index.tsx:44]: await Promise.all(dispatch(todoApi.util.getRunningQueriesThunk())

File: src/pages/todos/index.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    res.setHeader(
+      'Cache-Control',
+      'public, s-maxage=60, stale-while-revalidate=59',
+    );
+
+    const i18nProps = await serverSideTranslations(locale, ['common']);
+
+    return {
+      props: {
+        todos: data,
```
Comment (DmitriyGo) [inline src/pages/todos/index.tsx:55]: it's redundant

File: src/store/store.ts [hunk 5]
```diff
@@ -1,15 +1,43 @@
...
+  } else {
+    // we need it only on client side
+    const persistConfig = {
+      key: 'nextjs',
+      whitelist: ['auth'], // make sure it does not clash with server keys
+      storage,
+    };
+    const persistedReducer = persistReducer(persistConfig, rootReducer);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const store: any = configureStore({
```
Comment (DmitriyGo) [inline src/store/store.ts:32]: Is there a way to get rid of 'any'?

File: src/pages/todos/index.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import { GetServerSideProps } from 'next';
+import Head from 'next/head';
+import Link from 'next/link';
+import { serverSideTranslations } from 'next-i18next/serverSideTranslations';
+import styled from 'styled-components';
+
+import ToDoItem from './components/ToDoItem';
+
+import { selectTheme } from '@/modules/_Theme/feature/selectors';
```
Comment (DmitriyGo) [inline src/pages/todos/index.tsx:9]: It should be imported from module

File: src/pages/todos/components/ToDoItem.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  display: flex;
+  padding: 5px;
+  gap: 10px;
+  border: 5px solid ${(props) => props.theme.colorPrimary};
```
Comment (turchin-aa) [inline src/pages/todos/components/ToDoItem.tsx:22]: you can use destructure here ${({ theme }) => theme.colorPrimary}

File: src/modules/ToDo/feature/selectors.ts [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+};
+
+const ThemingSelector = (state: RootState): ToDoState => state[TODO_API_NAME];
+
+export const selectToDos = createSelector(
+  ThemingSelector,
+  (state) => state.todos,
+);
+
+export const { useGetToDosQuery } = todoApi;
```
Comment (turchin-aa) [inline src/modules/ToDo/feature/selectors.ts:17]: Why is it here?

Comment (turchin-aa) [inline src/modules/ToDo/feature/selectors.ts:17]: Why is this export here?

File: src/modules/ToDo/feature/todoApi.ts [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+export const todoApi = createApi({
+  reducerPath: TODO_API_NAME,
+  baseQuery: fetchBaseQuery({ baseUrl: 'https://dummyjson.com/todos/' }),
+  endpoints: (builder) => ({
+    getToDos: builder.query<ToDo[], void>({
+      query: () => '',
+      transformResponse: (res: ToDoResponce) => {
+        return res.todos;
+      },
```
Comment (turchin-aa) [inline src/modules/ToDo/feature/todoApi.ts:13]: you use different styles of arrow functions within the same scope

File: src/pages/todos/index.tsx [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+import { ToDo, todoApi } from '@/modules/ToDo';
+import { useSelector, wrapper } from '@/store';
+
+const ToDo = ({ todos }: { todos: ToDo[] }) => {
```
Comment (turchin-aa) [inline src/pages/todos/index.tsx:13]: Create a separate interface here

Comment (DmitriyGo) [reply src/pages/todos/index.tsx:13]: In fact it should be removed at all

**Review by turchin-aa — APPROVED** [review]

**Pull request merged**